phys/module_sf_noahlsm.F

```diff
@@ -4124,7 +4124,7 @@ SUBROUTINE TDFCND ( DF, SMC, QZ, SMCMAX, SH2O, BEXP, PSISAT, SOILTYP, OPT_THCND)
       IMPLICIT NONE
       INTEGER, INTENT(IN)       :: SOILTYP, OPT_THCND  
       REAL, INTENT(IN)          :: QZ,  SMC, SMCMAX, SH2O, BEXP, PSISAT
-      REAL, INTENT(OUT)         :: DF
+      REAL, INTENT(INOUT)       :: DF
```

Comment:
For all of the variables that are changed from (OUT) to (INOUT), can we check with Laura to see she found some problem with them?

Reply:
@weiwangncar 
If I changed them in the WRF code, it was because of MPAS code. Laura has made some suggestions for some to be changed to INOUT from IN/OUT with the MPAS code, but not the other way around. I have updated all files with her suggested changes for both WRF and MPAS PR's.

Reply:
I'd like to understand why the declaration of many variables have to be changed in MPAS.

Reply:
@weiwangncar 
Just want to update since it's been a while: Michael and Laura have looked into this but have not been able to find any record for why this was changed. There are theories, but they say they will still try to dig deeper.